databricks/koalas/data_type_ops/num_ops.py

```diff
@@ -0,0 +1,197 @@
+
+    def __radd__(self, left, right=None):
+        if isinstance(right, str):
+            raise TypeError("string addition can only be applied to string series or literals.")
+        return column_op(Column.__radd__)(left, right)
+
+    def __rsub__(self, left, right=None):
+        if isinstance(right, str):
+            raise TypeError("substraction can not be applied to string series or literals.")
+        return column_op(Column.__rsub__)(left, right)
+
+    def __rmul__(self, left, right=None):
+        if isinstance(right, str):
+            raise TypeError("multiplication can not be applied to a string literal.")
+        return column_op(Column.__rmul__)(left, right)
+
+    def __rtruediv__(self, left, right=None):
+        if isinstance(right, str):
+            raise TypeError("division can not be applied on string series or literals.")
+
+        def rtruediv(left, right):
+            return F.when(left == 0, F.lit(np.inf).__div__(right)).otherwise(
+                F.lit(right).__truediv__(left)
+            )
+
+        return numpy_column_op(rtruediv)(left, right)
+
+    def __rfloordiv__(self, left, right=None):
+        if isinstance(right, str):
+            raise TypeError("division can not be applied on string series or literals.")
+
+        def rfloordiv(left, right):
+            return F.when(F.lit(left == 0), F.lit(np.inf).__div__(right)).otherwise(
+                F.when(F.lit(left) == np.nan, np.nan).otherwise(F.floor(F.lit(right).__div__(left)))
+            )
+
+        return numpy_column_op(rfloordiv)(left, right)
+
+    def __rpow__(self, left, right=None):
+        if isinstance(right, str):
+            raise TypeError("division can not be applied on string series or literals.")
+
+        def rpow_func(left, right):
+            return F.when(F.lit(right == 1), right).otherwise(Column.__rpow__(left, right))
+
+        return column_op(rpow_func)(left, right)
+
+    def __rmod__(self, left, right=None):
+        if isinstance(right, str):
+            raise TypeError("modulo can not be applied on string series or literals.")
+
+        def rmod(left, right):
+            return ((right % left) + left) % left
+
+        return column_op(rmod)(left, right)
+
+
+class IntegralOps(NumericOps):
+    """
+    The class for binary operations of Koalas objects with spark types: LongType, IntegerType,
+    ByteType, and ShortType.
+    """
+
+    def __mul__(self, left, right):
+        if isinstance(right, str):
+            raise TypeError("multiplication can not be applied to a string literal.")
+
+        if isinstance(right, IndexOpsMixin) and isinstance(right.spark.data_type, StringType):
+            return column_op(SF.repeat)(right, left)
+        return column_op(Column.__mul__)(left, right)
+
+
+class FractionalOps(NumericOps):
+    """
+    The class for binary operations of Koalas objects with spark types: FloatType and DoubleType.
+    """
+
+    pass
+
+
+class DecimalOps(FractionalOps):
```

Comment:
@ueshin Would you show examples of why we wanted to make `FractionalOps` and `DecimalOps` each an individual class? Thanks!

Reply:
`DoubleType` and `FloatType` could have `NaN`, whereas `DecimalType` not.

Reply:
Wouldn't `column_op(Column.__mul__)(left, right)` be able to handle `NaN`?

I don't see why they cannot reuse `IntegralOps`.

Reply:
Removed DecimalOps for now.